packages/stringify/src/index.js

```diff
@@ -0,0 +1,134 @@
+import { toKebabCase } from './toKebabCase.js'
+import { getResolvedSelectors } from './getResolvedSelectors.js'
+
+/** Set used to manage the opened and closed state of rules. */
+const state = new WeakSet()
+
+/** Comma matcher outside rounded brackets. */
+const comma = /\s*,\s*(?![^()]*\))/
+
+/** Returns a string of CSS from an object of CSS. */
+export const stringify = (
+	/** Object representing the current CSS. */
+	value,
+	/** ... */
+	replacer = undefined,
+) => {
+	/** String of CSS being generated. */
+	let cssText = ''
+
+	/** Array of nesting conditions. */
+	const conditions = []
+
+	let lastName, lastData
+
+	/** Process CSS from an object of styles. */
+	const processStyle = (
+		/** Object representing the current CSS. */
+		style,
+
+		/** Styled rule selectors representing the current CSS. */
+		selectors,
+	) => {
+		for (let name in style) {
+			/** Whether the current style is a condition (i.e. media or supports query). */
+			const isCondition = name.charCodeAt(0) == 64
+			const isRule = isCondition ? !/^@(charset|import|namespace)$/.test(name) : /&/.test(name) || !selectors.length
```

Comment:
elegant and flexible way to handle things 🥇

sidenote: pretty sure `name.indexOf('&') != -1` would be much better perf-wise

Reply:
I really appreciate the review, especially from you, @arempe93.

I’m pleased to share that code golfing your suggestion to `~name.indexOf('&')` either breaks even or out-performs your `!= -1`, which greatly outperforms `/&/.test` just as you suggested.

Cite: https://jsben.ch/lcvO3

Reply:
Resolved in https://github.com/modulz/stitches/pull/477/commits/e2b0cf861262e228cd35a0fafc991e9cbee4b4ee

Testing `includes` yielded similar results as `indexOf`, so I landed there. Please feel free to comment anywhere things seem vague or troublesome.